enumerable_project.rb

```diff
@@ -0,0 +1,139 @@
+ Copyright (c) Muhammasd Arslan
+# frozen_string_literal: true
+
+module Enumerable
+  def my_each
+    return self unless block_given?
+
+    (0..length - 1).each do |i|
+      yield(self[i])
+    end
+    self
+  end
+
+  def my_each_with_index
+    return self unless block_given?
+
+    (0..length - 1).each do |i|
+      yield(self[i], i)
+    end
+  end
```

Comment:
my_each_with_index

Should returns an Enumerator if no block is given:
`p [1, 2, 3].my_each_with_index` returns `#<Enumerator>`